### ISSUE TITLE
SELECT FOR UPDATE/SHARE support

### ISSUE THREAD

**Issue by evilbloodydemon** [linq2db/linq2db#1276]
This is question/feature request - are there any plans for supporting 'SELECT ... FOR UPDATE/SHARE/ETC' query syntax? It's available in many RDBMS and it would be nice to have control over it.

**Comment by MaceWindu** [comment]
We already have very basic support for stuff like that in `IDataContext`:
```cs
		/// <summary>
		/// Gets list of query hints (writable collection), that will be used for all queries, executed using current context.
		/// </summary>
		List<string>        QueryHints        { get; }
		/// <summary>
		/// Gets list of query hints (writable collection), that will be used only for next query, executed using current context.
		/// </summary>
		List<string>        NextQueryHints    { get; }
```
which allow you to append and [since v2.1 prepend](https://github.com/linq2db/linq2db/wiki/Releases-and-Roadmap#prefix-query-hints-support) arbitrary sql to your all/next query.

But I agree that it is an area where we can have some improvements. Still no any specific plans what and how could be improved here.

**Comment by vanbukin** [comment]
+1 for that issue

**Comment by jods4** [comment]
@MaceWindu I need to use `FOR UPDATE OF T.column`
Until #2452 is done and this has nice proper support, is there a way to know in advance or force what a table alias will be?

**Comment by MaceWindu** [comment]
Don't think it is possible now. @sdanyliv ?

**Comment by sdanyliv** [comment]
Nope, not yet. Only by intercepting commands

**Comment by MaceWindu** [comment]
Implemented by #3297 (will be released with next v4 preview)
